ls1intum/Artemis
# `Quiz exercises`: Fix view submission for batch mode

### Checklist
#### General


#### Server

### Motivation and Context



[#5908](https://github.com/ls1intum/Artemis/issues/5908)

### Description


The error happens when a student has submitted answers for a batch quiz exercise and would like to view his/her submission before the quiz is ended. When viewing a submission, the server needs to find the quiz batch that the student participated in.

### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Student
- 1 Course

1. Log in to Artemis as the Instructor
2. Navigate to Course Administration
3. Select the Course
4. Click Exercises
5. Click Create Quiz
6. Set the Title
7. Change Batch Mode to **Batched**
8. Add any valid question(s)
9. Click Save
10. Click Back
11. Click Add Batch
12. Click Start Quiz
13. Copy / remember Password
14. Log Out and Log In as the Student
15. Select the Course
16. Click Exercises Tab
17. Click Open Quiz
18. Enter the Password and Click Join
19. Answer the questions
20. Click Submit
21. Click the Exercise Title of the navigation breadcrumb
22. Click View Submission
23. Verify that submitted answers are displayed
24. Log Out and Log In as the Instructor
25. End the quiz.
24. Log Out and Log In as the Student
25. Select the Course
26. Click Exercises Tab
27. Click View Result
28. Verify that submitted answers and results are displayed

### Review Progress






#### Code Review
#### Manual Tests

### Test Coverage



| Class/File | Branch | Line |
|------------|-------:|-----:|
| QuizBatch.java | 93% | 99% |
| QuizExercise.java | 75% | 93% |
| QuizSubmission | 75% | 92% |
| QuizBatchRepository | n/a | 57% |
| QuizBatchService | 82% | 90% |
| QuizSubmissionService | 75% | 83% |
| QuizScheduleService | 72% | 89% |
| ParticipationResource | 68% | 90% |

### Screenshots



#### Before

https://user-images.githubusercontent.com/10885503/205901465-7f5cc02b-ff9b-415a-af54-056b73435769.mov

#### After

https://user-images.githubusercontent.com/10885503/205901537-c7e2ce2d-4503-4115-b79c-5fba77f8314e.mov
